yarn-project/simulator/src/public/abstract_phase_manager.ts

```diff
@@ -169,35 +165,37 @@ export abstract class AbstractPhaseManager {
       c => revertibleCallStack.findIndex(p => p.equals(c)) !== -1,
     );
 
+    const teardownCallStackLength = arrayNonEmptyLength(data.publicTeardownCallStack, f => f.isEmpty());
+
+    const teardownCallStack = teardownCallStackLength === 0 ? [] : [tx.publicTeardownFunctionCall];
```

Comment:
Can it be `const teardownCallStack = !tx.publicTeardownFunctionCall.isEmpty() ? [tx.publicTeardownFunctionCall] : []`?
Using the length to create the stack makes it look like something is wrong: the length might be larger than 1 (although it won't be) but the array will only contain at most 1 item.